### ISSUE TITLE
Off-topic: MailExtension port for "Smart Quotes" extension

### ISSUE THREAD

**Issue by spodermenpls** [cleidigh/EditEmailSubject-MX#36]
This request isn't directly related to "EditEmailSubject", but since I couldn't find a way to contact @cleidigh directly, opening an 'issue' in this repo seemed to be the best alternative - feel free to move it or to name a better platform for discussion, if it's too 'out of line' in this location.

My extension of concern is a "conversion project" as well, it was published in 2013 and doesn't match the MailExtension architecture required by the latest Thunderbird versions: https://addons.thunderbird.net/en-US/thunderbird/addon/smart-quotes/. It provided support for typographic quotes, which aren't natively supported by Thunderbird/Firefox. There's a Bugzilla ticket that asks to introduce such feature to be embedded in Firefox (which could be transferred to Thunderbird then too, eventually), but it's been idling without priority since May 2002: https://bugzilla.mozilla.org/show_bug.cgi?id=145765.

The Thunderbird add-on website says that "Smart Quotes" is only compatible with TB 17, a user at the German Thunderbird support forum was able to tweak the original .xpi-package to make it compatible with TB 68 (appendix of the post: https://www.thunderbird-mail.de/forum/thread/82896-typographische-anf%C3%BChrungszeichen-smart-quotes/?postID=450279#post450279), Thunderbird 78 requires a MailExtension rewrite now, though, which the board user can't provide, due to time and experience constraints. The original author of the add-on seems to be more of a "JavaScript generalist", without experience on Web-/MailExtensions, and I am not a developer of any kind, that's why capable and experienced MailExtension developers as @cleidigh, or other team members of this repo, may be able to provide the know-how needed for this.

The existing extension code seems rather compact, so I'd be interested to hear an opinion on how much effort would be roughly required to transition its functionality into a MailExtension (which could also be enhanced by support for "Guillemets" for French texts, potentially). I know that many (non-English-writing) users, such as myself, would benefit from automatically inserted 'proper' quote symbols, without having to paste those one-by-one by hand. The lack of "Smart Quote" support is one of the very last shortcomings (from a non-corporate user's view, at least) of Thunderbird in comparison to Outlook that's not addressed in the TB Roadmap, recent TB beta versions or by other add-on developers, yet. I'd also be willing to support such a undertaking with a donation, if it'd make it more worthwhile for someone to spend time on rewriting efforts. 😄

**Comment by cleidigh** [comment]
@spodermenpls 
I happened to just see your post
I'm happy to help
if the extension runs correctly under 68 then you will probably not have much of a problem if all
utilizing the window listener API
Christopher

**Comment by eyalroz** [comment]
Luckily, and at least for the time being - you don't actually need to convert to being a MailExtension. That is, like Christopher suggests, you can use John Bieling's Extension Loader, to have an extension which basically loads your old extension's XUL, CSS and JS, and then it runs like it used to (well, up to the usual bits of work with every version change).

The loader is named "WindowListener" and you can check out the [guide](https://github.com/thundernest/addon-developer-support/wiki/Using-the-WindowListener-API-to-convert-a-Legacy-Overlay-WebExtension-into-a-MailExtension-for-Thunderbird-78) for using it.

Also, there's the Thunderbird Add-Ons mailing list / topicbox mechanism:
https://thunderbird.topicbox.com/groups/addons/

where you can ask questions which aren't "off topic issues".

Good luck!

**Comment by spodermenpls** [comment]
Thanks to both of you for pointing me to the "WindowListener" tool, which I didn't hear of before, that would avoid the need to re-write from scratch. I gave it a shot, unfortunately when it comes to transitioning resources from "JAR" files, porting XUL to XHTML or JS files and registering those correctly (also since the instructions aren't tailored for my specific use case, obviously), the needed practice exceeds my 'expertise' by a mile. I know now though how a port could be accomplished, and maybe I can persuade the couple of WebExtension devs I know, to lend me a hand with this. Thanks for your time!

**Comment by jobisoft** [comment]
@spodermenpls : Do you have your code in a public repository? I could have a look and send you a PR to get you started.

**Comment by eyalroz** [comment]
@spodermenpls : Well, you should probably look at my own transition of removedupes or bidimailui (here on GitHub) - because I did these things for my extensions. Also, "XUL to XHTML" is just renaming files, plus a few minor changes appearing as a few items on  [this list](https://developer.thunderbird.net/add-ons/updating/tb78/changes).

**Comment by spodermenpls** [comment]
@jobisoft Thanks for the offer. The original author has no repo for the extension that I could fork, so I've made one of my own (https://github.com/spodermenpls/smartquotes) and uploaded the original and my "partially" transitioned build in separate folders. I've added you as a collaborator, so you have viewing and writing rights, so feel free to take a look at it whenever you have some time hand.

@eyalroz Thanks for the hint. I've taken a look at those, your extensions are a lot more complex than the one I want to transition though, so isolating the applicable changes from your projects to mine is another hurdle for a total WebExtension/"Window Listener" newbie. I may try to climb the learning curve in the future, but I'll wait for jobisoft's informed opinion on my little project first, before I bang my head against the wall unnecessarily.
